outlines/generate/json.py

```diff
@@ -68,6 +69,35 @@ def json(
     return generator
 
 
+@regex.register(LlamaCpp)
+def json_llamacpp(
+    model,
+    schema_object: Union[str, object, Callable],
+    sampler: Sampler = multinomial(),
+):
+    if isinstance(schema_object, type(BaseModel)):
+        schema = pyjson.dumps(schema_object.model_json_schema())
+        regex_str = build_regex_from_object(schema)
+    elif callable(schema_object):
+        schema = pyjson.dumps(get_schema_from_signature(schema_object))
+        regex_str = build_regex_from_object(schema)
+    elif isinstance(schema_object, str):
+        schema = schema_object
+        regex_str = build_regex_from_object(schema)
+    else:
+        raise ValueError(
+            f"Cannot parse schema {schema_object}. The schema must be either "
+            + "a Pydantic object, a function or a string that contains the JSON "
+            + "Schema specification"
+        )
+
+    # TODO: format the output
+    # We should be able to use the same interface as transformers and make this
+    # function redundant by adding a `format_sequence` method to `LlamaCpp`
+
+    return regex(model, regex_str, sampler)
```

Comment:
Yes, we should add a `format_sequence` method to `LlamaCpp`.

Reply:
It is implemented but probably got mixed up when fixing conflicts.